File: spring/src/main/java/com/hcsy/spring/api/controller/SqlToolsController.java
package com.hcsy.spring.api.controller;

import java.util.List;
import java.util.Map;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import com.hcsy.spring.api.service.SqlToolsService;
import com.hcsy.spring.common.utils.Result;
import com.hcsy.spring.core.annotation.ApiLog;
import com.hcsy.spring.core.annotation.RequireInternalToken;
import com.hcsy.spring.entity.dto.SqlQueryDTO;

import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import reactor.core.publisher.Mono;

/**
 * SQL工具接口（供FastAPI Agent远程调用）
 * 提供受限的只读参数化SQL查询能力，表名白名单控制
 */
@RestController
@RequestMapping("/sql-tools")
@RequiredArgsConstructor
@Tag(name = "SQL工具", description = "供FastAPI Agent远程调用的SQL工具接口")
public class SqlToolsController {

    private final SqlToolsService sqlToolsService;

    @GetMapping("/tables")
    @Operation(summary = "获取表结构信息（内部）", description = "获取白名单内MySQL表的结构信息，供FastAPI Agent远程调用")
    @RequireInternalToken
    @ApiLog("内部获取表结构信息")
    public Mono<Result<List<Map<String, Object>>>> getTables(
        @RequestParam(required = false) String table) {
        return sqlToolsService.getTables(table).map(Result::success);
    }

    @PostMapping("/query")
    @Operation(summary = "执行只读SQL查询（内部）", description = "执行受限的只读参数化SQL查询，供FastAPI Agent远程调用")
    @RequireInternalToken
    @ApiLog("内部执行SQL查询")
    public Mono<Result<Map<String, Object>>> executeQuery(
        @Valid @RequestBody SqlQueryDTO dto) {
        return sqlToolsService.executeQuery(dto.getQuery(), dto.getParams())
            .map(Result::success);
    }
}

File: nestjs/src/module/system/sqlTools/sqlTools.controller.ts
import { Body, Controller, Get, Post, Query } from "@nestjs/common";
import { ApiOperation, ApiQuery, ApiTags } from "@nestjs/swagger";
import { success } from "src/common/utils/response";
import { ApiLog } from "src/framework/decorators/apiLog.decorator";
import { RequireInternalToken } from "src/framework/decorators/requireInternalToken.decorator";
import { SqlQueryDto } from "./dto/sqlQuery.dto";
import { SqlToolsService } from "./sqlTools.service";

@ApiTags("SQL工具")
@Controller("sql-tools")
export class SqlToolsController {
  constructor(private readonly sqlToolsService: SqlToolsService) {}

  @Get("tables")
  @ApiOperation({
    summary: "获取表结构信息",
    description:
      "查询白名单内 MySQL 表的结构信息，不传表名时返回所有白名单表列表，供内部服务远程调用",
  })
  @ApiQuery({
    name: "table",
    required: false,
    description: "表名，为空则返回所有白名单表",
  })
  @RequireInternalToken()
  @ApiLog("获取SQL工具表结构信息")
  async getTables(@Query("table") table?: string) {
    const data = await this.sqlToolsService.getTables(table);
    return success(data);
  }

  @Post("query")
  @ApiOperation({
    summary: "执行只读参数化SQL",
    description:
      "在白名单表范围内执行参数化只读 SQL 查询并返回结果，供内部服务远程调用",
  })
  @RequireInternalToken()
  @ApiLog("执行SQL工具只读查询")
  async executeQuery(@Body() dto: SqlQueryDto) {
    const data = await this.sqlToolsService.executeQuery(
      dto.query,
      dto.params ?? {},
    );
    return success(data);
  }
}
